src/gval/utils/loading_datasets.py

```diff
@@ -229,6 +238,200 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
         return xr_object
 
 
+def _get_raster_band_nodata(band_metadata, nodata_fill) -> Number:
+    """
+
+    Parameters
+    ----------
+    band_metadata : list
+        Metadata from raster:bands extension
+    nodata_fill : Number
+        Fill in value for missing data
+
+    Returns
+    -------
+    Number
+        Number representing nodata
+    """
+
+    if band_metadata:
+        prop_string = str(band_metadata.coords["raster:bands"].values)
+        idx1, idx2 = prop_string.find("{"), prop_string.rfind("}")
+
+        return ast.literal_eval(prop_string[idx1 : idx2 + 1]).get("nodata")
+    else:
+        if nodata_fill is None:
+            raise ValueError(
+                "Must have nodata fill value if nodata is not present in metadata"
+            )
+
+        return nodata_fill
+
+
+def _set_nodata(
+    stack: xr.DataArray, band_metadata: list = None, nodata_fill: Number = None
+) -> Number:
+    """
+
+    Parameters
+    ----------
+    stack : xr.DataArray
+        Data to set nodata attribute
+    band_metadata : list
+        Metadata from raster:bands extension
+    nodata_fill : Number
+        Fill in value for missing data
+
+    """
+
+    if stack.rio.nodata is not None:
+        stack.rio.write_nodata(stack.rio.nodata, inplace=True)
+    elif stack.rio.encoded_nodata is not None:
+        stack.rio.write_nodata(stack.rio.encoded_nodata, inplace=True)
+    else:
+        stack.rio.write_nodata(
+            _get_raster_band_nodata(band_metadata, nodata_fill), inplace=True
+        )
+
+
+def _set_crs(stack: xr.DataArray, band_metadata: list = None) -> Number:
+    """
+
+    Parameters
+    ----------
+    stack : xr.DataArray
+        Original data with no information
+    band_metadata : dict
+        Information with band metadata
+
+    Returns
+    -------
+    Xarray DataArray with proper CRS
+
+    """
+
+    if stack.rio.crs is not None:
+        return stack.rio.write_crs(stack.rio.crs)
+    else:
+        return stack.rio.write_crs(f"EPSG:{band_metadata['epsg'].values}")
+
+
+def get_stac_data(
```

Comment:
Should we be returning stac's as catalogs or xarray's? I believed the original thought was to return catalogs in the panda's DF style?

Or is this the lower level function and some future function will turn filtered STAC objects into catalog DF's?

Reply:
This particular function was made to get an online asset directly in to an xarray object.  In the event that a catalog with STAC items was created as you are referring to, then at process time this function would be able to create xarray objects for each asset in the DataFrame and then do the actual comparisons.